jgrapht-core/src/test/java/org/jgrapht/alg/decompostion/NiceDecompositionBuilderTestUtil.java

```diff
@@ -0,0 +1,115 @@
+{
+    //static class
+    private NiceDecompositionBuilderTestUtil() {}
+    
+    /**
+     * tests whether the tree decomposition (decomposition, map) with root is nice
+     * 
+     * @param decomposition the tree decomposition to check
+     * @param map the map of the tree decomposition
+     * @param root the root of the tree decomposition
+     */
+    public static <V,E,W> void testNiceDecomposition(Graph<V,E> decomposition, Map<V,Set<W>> map, V root){
+        
+        Queue<V> queue = new LinkedList<V>();
+        
+        //test and add root
+        assertTrue(root+" is no valid root"
+                + "\n in decomposition "+decomposition
+                + "\n and map"+map, map.get(root).isEmpty());
+       queue.add(root);
+        
+        while(!queue.isEmpty())
+        {
+            V current = queue.poll();
+            List<V> successor = Graphs.successorListOf(decomposition, current);
+            if(successor.size() == 0 && map.get(current).isEmpty()) continue; //leaf node
+            if(successor.size() == 1) //forget or introduce
+            {
+                V next = successor.get(0);
+                queue.add(next);
+                Set<W> union = new HashSet<W>(map.get(current));
+                union.addAll(map.get(next));
+                if(union.size() == map.get(next).size() || union.size() == map.get(current).size())
+                {
+                    if(map.get(current).size() == map.get(next).size()-1) continue; //introduce
+                    else if(map.get(current).size()-1 == map.get(next).size()) continue; //forget
+                }
+            }
+            if(successor.size() == 2) //join
+            {
+                V first = successor.get(0);
+                V second = successor.get(1);
+                queue.add(first);
+                queue.add(second);
+                Set<W> union = new HashSet<W>(map.get(current));
+                union.addAll(map.get(first));
+                union.addAll(map.get(second));
+                if(union.size() == map.get(current).size() 
+                && union.size() == map.get(first).size() 
+                && union.size() == map.get(second).size()) 
+                    continue; //join node!
+            }
+            assertFalse("Vertex Set "+current+" is not a valid node for a nice decomposition"
+                + "\nin decomposition "+decomposition
+                + "\nwith map "+map, true); //no valid node!
+        }
+        assertTrue(true);
+    }
+    
+    /**
+     * Test whether (decomposition, map) a tree decomposition of oldGraph is
+     * 
+     * @param oldGraph the graph of the tree decomposition
+     * @param decomposition the tree decomposition
+     * @param map the map from vertices to the tree decomposition to the sets of the tree decomposition
+     */
+    public static <V,E,W,F> void testDecomposition(Graph<W,F> oldGraph, Graph<V,E> decomposition, Map<V,Set<W>> map){
```

Comment:
As a general idea, you can make all these functions return a boolean and use `Assert.assertTrue/False` in the actual tests. This way you can test both cases. Your way can only be used to test whether A is B. It doesn't work to test if A is not B.